pkg/rancher-desktop/config/settings.ts

```diff
@@ -317,39 +352,78 @@ export function updateFromCommandLine(cfg: Settings, commandLineArgs: string[]):
 
   return cfg;
 }
+
 /**
  * Load the settings file or create it if not present.  If the settings have
  * already been loaded, return it without re-loading from disk.
  */
 export function load(): Settings {
+  let deploymentProfiles: DeploymentProfileType = { defaults: {}, locked: {} };
+  let setDefaultMemory = false;
+  let lockedProfileSettings: RecursivePartial<Settings>;
+
+  try {
+    deploymentProfiles = readDeploymentProfiles();
+    lockedProfileSettings = deploymentProfiles.locked;
+  } catch {
+    lockedProfileSettings = {};
+  }
   try {
     settings ??= loadFromDisk();
   } catch (err: any) {
     settings = clone(defaultSettings);
     if (err.code === 'ENOENT') {
-      _isFirstRun = true;
-      if (os.platform() === 'darwin' || os.platform() === 'linux') {
-        const totalMemoryInGB = os.totalmem() / 2 ** 30;
-
-        // 25% of available ram up to a maximum of 6gb
-        settings.virtualMachine.memoryInGB = Math.min(6, Math.round(totalMemoryInGB / 4.0));
+      if (Object.keys(deploymentProfiles.defaults).length) {
+        _.merge(settings, deploymentProfiles.defaults);
+        if (!_.has(deploymentProfiles.defaults, 'virtualMachine.memoryInGB')) {
+          setDefaultMemory = true;
+        }
+      } else {
+        _isFirstRun = true;
```

Comment:
This means it's not possible to provide defaults _and_ trigger the first run.

For the first go that's probably okay.

Reply:
This is intentional (#3759):

> If either a system or user deployment profile exists, then the first run dialog should not be displayed, and any missing values get the default value (same as pressing "Accept" in the first-run dialog without modifying any settings).

We can argue if this is the right behaviour, but mixing deployment profiles with a first-run dialog gets kind of complicated. And first-run is just  an optimization; the user can always change the settings in Preferences again right away.

Reply:
Man this is subtle. "[The defaults part of the deployment profile] provides the initial first-run configuration if settings.json does not yet exist. If all required settings are provided, then no first-run dialog will be shown."

Yeah, let me get that in.